pkg/volume/volume.go

```diff
@@ -0,0 +1,316 @@
+package volume
+
+import (
+	"context"
+	"fmt"
+
+	volumeapi "github.com/kubernetes-csi/csi-proxy/pkg/volume/api"
+	"k8s.io/klog/v2"
+)
+
+// Volume wraps the host API and implements the interface
+type Volume struct {
+	hostAPI volumeapi.API
+}
+
+type Interface interface {
+	DismountVolume(context.Context, *DismountVolumeRequest) (*DismountVolumeResponse, error)
+	FormatVolume(context.Context, *FormatVolumeRequest) (*FormatVolumeResponse, error)
+	GetClosestVolumeIDFromTargetPath(context.Context, *GetClosestVolumeIDFromTargetPathRequest) (*GetClosestVolumeIDFromTargetPathResponse, error)
+	GetDiskNumberFromVolumeID(context.Context, *GetDiskNumberFromVolumeIDRequest) (*GetDiskNumberFromVolumeIDResponse, error)
+	GetVolumeDiskNumber(context.Context, *VolumeDiskNumberRequest) (*VolumeDiskNumberResponse, error)
+	GetVolumeIDFromMount(context.Context, *VolumeIDFromMountRequest) (*VolumeIDFromMountResponse, error)
+	GetVolumeIDFromTargetPath(context.Context, *GetVolumeIDFromTargetPathRequest) (*GetVolumeIDFromTargetPathResponse, error)
+	GetVolumeStats(context.Context, *GetVolumeStatsRequest) (*GetVolumeStatsResponse, error)
+	IsVolumeFormatted(context.Context, *IsVolumeFormattedRequest) (*IsVolumeFormattedResponse, error)
+	ListVolumesOnDisk(context.Context, *ListVolumesOnDiskRequest) (*ListVolumesOnDiskResponse, error)
+	MountVolume(context.Context, *MountVolumeRequest) (*MountVolumeResponse, error)
+	ResizeVolume(context.Context, *ResizeVolumeRequest) (*ResizeVolumeResponse, error)
+	UnmountVolume(context.Context, *UnmountVolumeRequest) (*UnmountVolumeResponse, error)
+	VolumeStats(context.Context, *VolumeStatsRequest) (*VolumeStatsResponse, error)
```

Comment:
Since this API will be for new clients and maintaining perfect compat with prior APIs is not as big of a goal, how about dropping `VolumeStats` (since we already have `GetVolumeStats`)?

Reply:
tracked in https://github.com/kubernetes-csi/csi-proxy/issues/241